### PR TITLE
Enable clang-tidy CI checks on private branches

### PR DESCRIPTION
**Pull request by nilsvu**
## Proposed changes

For pull requests we check code that changed relative to the base
branch and for pushes to private branches we check code that changed
relative to the repository's `develop` branch.

### Types of changes:

- [ ] Bugfix
- [x] New feature
- [ ] Refactor

### Component:

- [ ] Code
- [ ] Documentation
- [ ] Build system
- [x] Continuous integration

### Code review checklist

- [ ] The PR passes all checks, including unit tests and `clang-tidy`.
  For instructions on how to perform the CI checks locally refer to the [Dev
  guide on the Travis CI](https://spectre-code.org/travis_guide.html).
- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).

### CODE REVIEW

**Comment by nilsvu** [comment]
@kidder @nilsdeppe this is ready for you to review whenever you get the chance

**Comment by kidder** [comment]
@nilsleiffischer can you push a change that removes a nolint so I can see clang-tidy failing CI on your branch...

**Comment by kidder** [comment]
you can squash

File: .github/workflows/Tests.yaml [hunk 1]
```diff
@@ -75,10 +75,13 @@ jobs:
           cd $GITHUB_WORKSPACE
           ./tools/CheckFiles.sh
 
-  # Lint the code that changed in a pull request with clang-tidy.
+  # Lint with clang-tidy. For pull requests we check code that changed relative
+  # to the base branch and for pushes we check code that changed relative to the
```
Comment (nilsdeppe) [inline .github/workflows/Tests.yaml]: let's just compare to sxs-collaboration/develop. That's what Travis was doing and worked really well.

**Review by nilsdeppe — CHANGES_REQUESTED** [review]

**Review by nilsdeppe — COMMENTED** [review]
LGTM, please go ahead and squash

**Review by kidder — COMMENTED** [review]
yes, squash

**Review by kidder — APPROVED** [review]

**Comment by kidder** [comment]
@nilsdeppe I think this just needs your approval

**Review by nilsdeppe — APPROVED** [review]

**Pull request merged**